tfworker/commands/root.py

```diff
@@ -163,6 +168,32 @@ def get_config_var_dict(config_vars):
     return return_vars
 
 
+def replace_hcl_vars(hcl, args):
+    if isinstance(hcl, dict):
+        for key in hcl.keys():
+            hcl[key] = replace_hcl_vars(hcl[key], args)
+    elif isinstance(hcl, list):
+        if len(hcl) == 1:
+            return replace_hcl_vars(hcl[0], args)
+        for i in range(len(hcl)):
+            hcl[i] = replace_hcl_vars(hcl[i], args)
+    elif isinstance(hcl, str):
+        return replace_vars(hcl, args)
+    return hcl
+
+
+def ordered_config_load_hcl(stream, args):
+    """
+    Load a yaml config, and replace templated items.
+
+    Derived from:
+    https://stackoverflow.com/questions/5121931/in-python-how-can-you-load-yaml-mappings-as-ordereddicts
+    """
+    hcl2_ordered = Lark_StandAlone(transformer=hcl2_transformer.OrderedDictTransformer())
+    doc = hcl2_ordered.parse(stream + "\n")
+    return replace_hcl_vars(doc, args)
+
+
 def ordered_config_load(
     stream, args, Loader=yaml.SafeLoader, object_pairs_hook=OrderedDict
```

Comment:
does this also works for json?

Reply:
YAML is a superset of JSON, so it should just work.  The only issue I see is that pyyaml doesn't support YAML 1.2 which fixed things like `no` being `false` that JSON doesn't do.  I haven't had it fail yet, but I'll do some specific testing of the JSON loading.

Reply:
I did some testing and it seems to behave exactly as the same as when it's loading YAML.